### DISCUSSION TITLE
@sveltejs/kit/vite -- is this reasonable workaround

### DISCUSSION THREAD

**Discussion by bfmcneill** [fontsource/fontsource#993] [Q&A]
I installed granstander using pnpm and got some errors when I tried to import the font into my project.  The `fontsource` docs seemed pretty straight forward for integration with `sveltekit`.  The only way I could get it to work is very different than the docs.  Am I just noobin out here?

```bash
pnpm add -D @fontsource/grandstander
```

I then imported it into my `routes/+layout.svelte`.

I then got an error that grandstander was outsite Vite serving allow list

```bash
project/node_modules/.pnpm/@fontsource+grandstander@5.0.22/node_modules/@fontsource/grandstander/files/grandstander-latin-400-normal.woff2" is outside of Vite serving allow list.
```

warning message said these are the only files allowed to be served
```bash
projectsrc/lib
project/src/routes
project/.svelte-kit
project/src
project/node_modules
```

I was able to work around this by finding fontsource paths from a function and modifying vite.config.ts

```typescript
import { sveltekit } from '@sveltejs/kit/vite';
import { defineConfig } from 'vite';
import { fileURLToPath } from 'url';
import path from 'path';
import fs from 'fs';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

// required to allow fontsource to play nice with vite security
function getFontsourcePaths() {
  const pnpmDir = path.resolve(__dirname, '../node_modules/.pnpm');
  const fontsourceDirs = fs.readdirSync(pnpmDir)
    .filter(dir => dir.startsWith('@fontsource+'))
    .map(dir => path.join(pnpmDir, dir, 'node_modules/@fontsource'));
  return fontsourceDirs;
}

export default defineConfig({
  plugins: [sveltekit()],
  server: {
    fs: {
      allow: [
        ...getFontsourcePaths()
      ]
    }
  }
});
```

**Comment by ayuhito** [comment]
> pnpm add -D @fontsource/grandstander

Shouldn't it be a direct dependency, not a dev dependency?

There shouldn't really be a reason to use a workaround, so this might be related.
